hub/core/tensor.py

```diff
@@ -77,15 +74,17 @@ def add_samples_to_tensor(
         TensorDoesNotExistError: If a tensor at `key` does not exist. A tensor must be created first using `create_tensor(...)`.
     """
 
-    # TODO: split into `append` and `extend`
-
-    array = normalize_and_batchify_shape(array, batched=batched)
-
     if not tensor_exists(key, storage):
         raise TensorDoesNotExistError(key)
 
     index_map = read_index_map(key, storage)
     tensor_meta = read_tensor_meta(key, storage)
+
+    array = normalize_and_batchify_shape(array, batched=batched)
```

Comment:
We still have the problem of storing the shape after it's been normalized. So if we store something of shape (1024, 1, 1024) we get back something of shape (1024, 1024). 

If we want to change this, it's as simple as moving this line after the meta update and fixing the expectations in the tests.